addons/mass_mailing/tests/test_mailing_controllers.py

```diff
@@ -174,6 +174,38 @@ def test_mailing_report_unsubscribe(self):
         self.assertEqual(res.status_code, 200)
         self.assertFalse(self.env['ir.config_parameter'].sudo().get_param('mass_mailing.mass_mailing_reports'))
 
+    def test_mailing_private_list_not_in_my(self):
+        """ Test that only the proper mailing lists are shown in the 'my' and
+        the mailing lists which we are not members and are not public shouldn't
+        be in this list.
+        """
+
+        private_list = self.env['mailing.list'].with_context(self._test_context).create({
+            'contact_ids': [
+                (0, 0, {'name': 'Carvajal', 'email': 'carvajal@example.com'}),
+            ],
+            'name': 'List5',
+            'is_public': False
+        })
+        private_list.subscription_ids[0].opt_out = True
+
+        portal_user = mail_new_test_user(
+            self.env,
+            email=tools.formataddr(("Carvajal", "carvajal@example.com")),
+            groups='base.group_portal',
+            login='user_portal_fleurus',
+            name='Robert User',
+            signature='--\nRobert',
+        )
+
+        self.authenticate(portal_user.login, portal_user.login)
+        with freeze_time(self._reference_now):
+            self.start_tour(
+                "/mailing/my",
+                "test_mailing_private_list_not_in_my",
+                login=portal_user.login,
+            )
```

Comment:
Could we just update `test_mailing_unsubscribe_from_my` instead?

It's best to avoid adding new tours as the set-up and tear-down for tours is super expensive. For static pages parsing the lxml result from using self.open is also a good option (but here as there's already a tour let's try using that one)